addons/currency_rate_live/models/currency_rate_update.py

```diff
@@ -0,0 +1,34 @@
+from openerp import models, fields, api, _
+import json
+
+
+class Currency_rate_update(models.Model):
+    _name = "res.currency.rate.provider"
+
+    service_provider = fields.Selection([
+            ('yahoo_finance', 'Yahoo Finance'),
+            ('ecb', 'European Central Bank'),
+            ('bancico', 'Bank Of Maxico'),
+            ('bank_of_canada', 'Bank Of Canada'),
+            ('pln', 'Polish National Bank'),
+            ('swiss_custom', 'Swiss Custom Admin'),
+            ('appspot', 'AppSpot'),
+            ('quandl', 'Quandl'),
+        ], string='Select service provider', required=True,
+        default=lambda self: self._context.get('service_provider', 'yahoo_finance'))
+    currency_ids = fields.Many2many('res.currency', size=3, string="Currencies to be update", required=True)
+    company_id = fields.Many2one('res.company')
+    base_currency_id = fields.Many2one('res.currency', string="Base currency", size=3, required=True, help="Input Base Currency name", default=1)
+    request_token = fields.Char('Request token')
```

Comment:
You can remove the string parameter here, as the ORM will generate a capitalized version of the field name by default.